ext/include/opentelemetry/ext/http/common/url_parser.h

```diff
@@ -88,8 +89,11 @@ class UrlParser
       path_ = std::string("/");  // use default path
       if (is_port)
       {
-        port_ = static_cast<uint16_t>(
-            std::stoi(std::string(url_.begin() + cpos, url_.begin() + url_.length())));
+        std::string port_str(
+            url_.begin() + static_cast<std::string::difference_type>(cpos),
+            url_.begin() + static_cast<std::string::difference_type>(url_.length()));
```

Comment:
`static_cast<>` is because "Narrowing conversion from 'size_t' (aka 'unsigned long') to signed type 'difference_type' (aka 'long') is implementation-defined"

See also: https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/narrowing-conversions.html